diff -u ./lingeling-bbc-9230380-160707/lglib.c ./lingeling/lglib.c
--- ./lingeling-bbc-9230380-160707/lglib.c	2016-07-07 16:21:31.000000000 +0800
+++ ./lingeling/lglib.c	2016-09-16 00:06:04.424228471 +0800
@@ -25445,3 +25445,16 @@
   }
   return res;
 }
+
+int lglvisits(LGL * lgl)
+{
+  Stats * s = lgl->stats;
+  int visits = s->visits.search + s->visits.simp + s->visits.lkhd;
+  return visits;
+}
+
+double lglsearchtime(LGL * lgl)
+{
+  double st = lgl->times->search;
+  return st;
+}
diff -u ./lingeling-bbc-9230380-160707/lglib.h ./lingeling/lglib.h
--- ./lingeling-bbc-9230380-160707/lglib.h	2016-04-28 04:01:55.000000000 +0800
+++ ./lingeling/lglib.h	2016-09-16 00:07:07.732644357 +0800
@@ -338,4 +338,6 @@
 void lglsetmsglock (LGL *, void (*lock)(void*), void (*unlock)(void*), void*);
 void lglsetime (LGL *, double (*time)(void));
 
+int lglvisits(LGL * lgl);
+double lglsearchtime(LGL * lgl);
 #endif
